projects/fsastorefrontlib/src/cms-components/my-account/claim/create-claim/create-claim.component.ts

```diff
@@ -33,24 +33,27 @@ export class CreateClaimComponent implements OnInit, OnDestroy {
   }
 
   startClaim() {
+    if (!this.confirm) {
+      return;
+    }
     this.subscription.add(
       this.isPolicySelected$
         .pipe(
+          filter(policy => !!policy?.userId),
           map(policy => {
-            if (policy && policy.userId) {
-              this.claimService.createClaim(policy.policyId, policy.contractId);
-              this.fileUploadService.resetFiles();
-              this.routingService.go({
-                cxRoute: 'fnolIncidentPage',
-              });
-            }
+            this.claimService.createClaim(policy.policyId, policy.contractId);
+            this.fileUploadService.resetFiles();
+            this.routingService.go({
+              cxRoute: 'fnolIncidentPage',
+            });
           })
         )
         .subscribe()
     );
   }
 
   ngOnDestroy() {
+    this.claimService.resetSelectedPolicy();
```

Comment:
I found one more bug and fix it with this line of code. When user selects claim and checkbox on fnol page from homepage prevention of disabled button state will work. When user again visit this page and selects checkbox, but not the claim card, button will be enabled and he can proceed even though he didn't select the claim. So once user leaves the `fnol` page, we need to reset `isPolicySelected ` state and nullify it.